test/coverage-test/option-tests/thresholds-auto-update.test.ts

```diff
@@ -0,0 +1,162 @@
+import { createCoverageSummary } from 'istanbul-lib-coverage'
+
+import { expect, test } from 'vitest'
+import { defineConfig } from 'vitest/config'
+import { BaseCoverageProvider } from 'vitest/coverage'
+
+const initialThresholds = { lines: 1, branches: 2, functions: 3, statements: 4 }
+const coveredThresholds = { lines: 50, branches: 60, functions: 70, statements: 80 }
+const initialConfig = JSON.stringify(defineConfig({
+  test: {
+    coverage: {
+      thresholds: initialThresholds,
+    },
+  },
+}), null, 2)
+
+test('updates thresholds on "export default {..}"', async () => {
+  const config = parseModule(`export default ${initialConfig}`)
+
+  const updatedConfig = await updateThresholds(config)
+
+  expect(updatedConfig).toMatchInlineSnapshot(`
+    "export default {
+      "test": {
+        "coverage": {
+          "thresholds": {
+            "lines": 50,
+            "branches": 60,
+            "functions": 70,
+            "statements": 80
+          }
+        }
+      }
+    }"
+  `)
+})
+
+test('updates thresholds on "export default defineConfig({...})"', async () => {
+  const config = parseModule(`export default defineConfig(${initialConfig})`)
+
+  const updatedConfig = await updateThresholds(config)
+
+  expect(updatedConfig).toMatchInlineSnapshot(`
+    "export default defineConfig({
+      "test": {
+        "coverage": {
+          "thresholds": {
+            "lines": 50,
+            "branches": 60,
+            "functions": 70,
+            "statements": 80
+          }
+        }
+      }
+    })"
+  `)
+})
+
+test('updates thresholds on "export default defineConfig(() => ({...}))"', async () => {
+  const config = parseModule(`export default defineConfig(() => (${initialConfig}))`)
+
+  const updatedConfig = await updateThresholds(config)
+
+  expect(updatedConfig).toMatchInlineSnapshot(`
+    "export default defineConfig(() => ({
+      "test": {
+        "coverage": {
+          "thresholds": {
+            "lines": 50,
+            "branches": 60,
+            "functions": 70,
+            "statements": 80
+          }
+        }
+      }
+    }))"
+  `)
+})
+
+test('updates thresholds on "export default mergeConfig({...}, defineConfig({...}))"', async () => {
+  const config = parseModule(`export default mergeConfig(baseConfig, defineConfig(${initialConfig}))`)
+
+  const updatedConfig = await updateThresholds(config)
+
+  expect(updatedConfig).toMatchInlineSnapshot(`
+    "export default mergeConfig(baseConfig, defineConfig({
+      "test": {
+        "coverage": {
+          "thresholds": {
+            "lines": 50,
+            "branches": 60,
+            "functions": 70,
+            "statements": 80
+          }
+        }
+      }
+    }))"
+  `)
+})
+
+test('updates thresholds on "export default mergeConfig({...}, defineConfig(() => ({...})))"', async () => {
+  const config = parseModule(`export default mergeConfig(baseConfig, defineConfig(() => (${initialConfig})))`)
```

Comment:
This doesn't actually work at runtime:

https://vitest.dev/config/file.html#managing-vitest-config-file (last code sample)

Reply:
I've replaced this now with `export default defineConfig((configEnv) => mergeConfig(baseConfig, defineConfig({})))`.